File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min_p = prices[0]
        max_p = prices[0]
        curr_profit = 0
        for price in prices:
            if price < min_p:
                curr_profit = max(curr_profit, max_p-min_p)
                min_p = price
                max_p = price
            if price > max_p:
                max_p = price
        return max(curr_profit, max_p-min_p)

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minPrice = prices[0];
        int maxProfit = 0;
        for (int i = 1; i < prices.size(); i++){
            int price = prices[i];
            if (price < minPrice){
                minPrice = price;
            }
            else if(price - minPrice > maxProfit){
                maxProfit = price - minPrice;
            }
            }
        return maxProfit;
        }

};